include/mc_tasks/lipm_stabilizer/StabilizerTask.h

```diff
@@ -315,6 +301,167 @@ struct MC_TASKS_DLLAPI StabilizerTask : public MetaTask
+    c_.dcmIntegralGain = clamp(i, 0., c_.safetyThresholds.MAX_DCM_I_GAIN);
+    c_.dcmDerivGain = clamp(d, 0., c_.safetyThresholds.MAX_DCM_D_GAIN);
+  }
+
+  void dcmIntegratorTimeConstant(double dcmIntegratorTimeConstant)
+  {
+    c_.dcmIntegratorTimeConstant = dcmIntegratorTimeConstant;
+    dcmIntegrator_.timeConstant(dcmIntegratorTimeConstant);
+  }
+
+  void dcmDerivatorTimeConstant(double dcmDerivatorTimeConstant)
+  {
+    c_.dcmDerivatorTimeConstant = dcmDerivatorTimeConstant;
+    dcmDerivator_.timeConstant(dcmDerivatorTimeConstant);
+  }
+
+  void comWeight(double weight)
+  {
+    c_.comWeight = weight;
+    comTask->weight(weight);
+  }
+
+  void comStiffness(const Eigen::Vector3d & stiffness)
+  {
+    c_.comStiffness = stiffness;
+    comTask->stiffness(stiffness);
+  }
+
+  void contactWeight(double weight)
+  {
+    c_.contactWeight = weight;
+    for(auto footT : contactTasks)
+    {
+      footT->weight(c_.contactWeight);
+    }
+  }
+
+  void contactStiffness(const sva::MotionVecd & stiffness)
+  {
+    c_.contactStiffness = stiffness;
+    for(auto contactT : contactTasks)
+    {
+      contactT->stiffness(stiffness);
+    }
+  }
+
+  void contactDamping(const sva::MotionVecd & damping)
+  {
+    c_.contactDamping = damping;
+    for(auto contactT : contactTasks)
+    {
+      contactT->damping(damping);
+    }
+  }
+
+  void copAdmittance(const Eigen::Vector2d & copAdmittance)
+  {
+    c_.copAdmittance = clamp(copAdmittance, 0., c_.safetyThresholds.MAX_COP_ADMITTANCE);
+    for(auto contactT : contactTasks)
+    {
+      contactT->admittance(contactAdmittance());
+    }
+  }
+
+  void copMaxVel(const sva::MotionVecd & copMaxVel)
+  {
+    c_.copMaxVel = copMaxVel;
+    for(const auto & footTask : footTasks)
+    {
+      footTask.second->maxLinearVel(copMaxVel.linear());
+      footTask.second->maxAngularVel(copMaxVel.angular());
+    }
+  }
+
+  void vdcFrequency(double freq)
+  {
+    c_.vdcFrequency = clamp(freq, 0., 10.);
+  }
+
+  void vdcStiffness(double stiffness)
+  {
+    c_.vdcStiffness = clamp(stiffness, 0., 1e4);
+  }
+
+  void dfzAdmittance(double dfzAdmittance)
+  {
+    c_.dfzAdmittance = clamp(dfzAdmittance, 0., c_.safetyThresholds.MAX_DFZ_ADMITTANCE);
+  }
+
+  void dfzDamping(double dfzDamping)
+  {
+    c_.dfzDamping = clamp(dfzDamping, 0., c_.safetyThresholds.MAX_DFZ_DAMPING);
+  }
+
+  void fdqpWeights(const FDQPWeights & fdqp)
+  {
+    c_.fdqpWeights = fdqp;
+  }
+
+  /**
+   * @brief Changes the safety thresholds
+   *
+   * This ensures that all the parameters depending on these safety parameters
+   * are withing the new thresholds. If they are out of bounds, they will be
+   * clamped back to the new range, and a warning message will be displayed.
+   *
+   * @param thresholds New safety thresholds
+   */
+  void safetyThresholds(const SafetyThresholds & thresholds)
+  {
+    c_.safetyThresholds = thresholds;
+    c_.clampGains();
+    copAdmittance(c_.copAdmittance);
```

Comment:
A comment to say that this is only required because we want to apply the new gain immediately would be nice